swap/swap_test.go

```diff
@@ -165,20 +451,617 @@ func createTestSwap(t *testing.T) (*Swap, string) {
 	if err2 != nil {
 		t.Fatal(err2)
 	}
-	swap := New(stateStore)
+	log.Debug("creating simulated backend")
+
+	swap := New(stateStore, key, common.Address{}, defaultBackend)
+	defaultBackend.Commit()
 	return swap, dir
 }
 
+// create a test swap account with a backend
+// creates a stateStore for persistence and a Swap account
+// returns a cleanup function
+func newTestSwap(t *testing.T, key *ecdsa.PrivateKey) (*Swap, func()) {
+	t.Helper()
+	swap, dir := newBaseTestSwap(t, key)
+	clean := func() {
+		swap.backend = nil
```

Comment:
This is not needed. It can be removed as it uses a data race with handleEmitChequeMsg.

Reply:
@janos can you explain why? the relation between `newTestSwap` and `handleEmitChequeMsg` is not immediately clear to me.

Reply:
Field backend is read in the handler in its own gorutine while the same field is set in a different goroutine in the clean function. There is no synchronization between two goroutines so it is possible that this read and write operations on the same field can result in a data race. It is reported by race detector when running tests in this package.